### ISSUE TITLE
Floccus does not sync anymore since v5.4.0 -- Self-signed certificate?

### ISSUE THREAD

**Issue by hochl** [floccusaddon/floccus#1787]
### Which version of floccus are you using?

5.4.0

### How many bookmarks do you have, roughly?

500

### Are you using other means to sync bookmarks in parallel to floccus?

No

### Sync method

WebDAV

### Which browser are you using? In case you are using the phone App, specify the Android or iOS version and device please.

Firefox on Linux and Windows 10/11.

### What kind of WebDAV server are you using? (if relevant)

rclone

### Describe the Bug

Everything worked flawlessly, floccus version V5.3.4 following the information window. Now none of my 5 Browser installs sync anymore, version installed is V5.4.0.

### Expected Behavior

Obviously, all browsers should sync. The sync does not finish, thus i cannot export a debug log. On Linux I got an error message yesterday that said something with E030.

### To Reproduce

Upgrade to V5.4.0 and you are hosed.

### Debug log provided

- [ ] I have provided a debug log file

**Comment by developer91234** [comment]
Can confirm. I've never tried this extension before but in my setup with webdav it won't sync anything despite not reporting any errors. I have two browsers, each with a bookmarks folder named flocus. The first one contains some bookmarks, the second is empty. No matter how many times I sync, or how much I fiddle with the sync settings, the second remains empty: nothing is synced to the second browser.

**Comment by marcelklehr** [comment]
Hello,

Thank you for reaching out. I can not reproduce this offhand, sadly. Can you perhaps share a screenshot?

**Comment by marcelklehr** [comment]
Yeah, network error is unlikely to be a problem in floccus.

**Comment by qlifee** [comment]
I have the same issue with the same error code E017 on: 

1) Pixel 9 Android 15 (graphene os) Floccus 5.3.4.
Surface running both 
2) Linux Mint chromium browser Floccus 5.4.0 and 
3) windows 11 running chrome browser Floccus 5.4.0.

This is my first time using Floccus and i didn't manage to get it to work.

I have WEBDAV server up and running on my Synology NAS, and i verified that it is working alright with various applications.

I suspect the problem is that Floccus doesn't allow self-signed certificates (No option to allow that), because the applications that didn't work are the ones that doesn't allow the self-signed certificates.

my WebDAV URL is `https://<tailscale-ip-address>:<port>/Bookmarks`
File format: XBEL

Note: I use tailscale to access my NAS.

![Screenshot from 2024-12-10 19-06-25](https://github.com/user-attachments/assets/25b101c1-64c6-4bee-93d6-0f000744a3fd)
![Screenshot_20241210-190709](https://github.com/user-attachments/assets/564b710f-9d4a-4b86-a5c3-160250113305)

**Comment by marcelklehr** [comment]
> self-signed certificates

Yes, that's a problem. In the browser you can accept the certificate manually, but on android there's currently no way to import the cert in a way that will make floccus sync, sadly.

**Comment by qlifee** [comment]
> > In the browser you can accept the certificate manually.

I didn't get prompted to accept a certificate, how can i do that?

**Comment by marcelklehr** [comment]
> I didn't get prompted to accept a certificate, how can i do that?

You can try accessing the webdav endpoint manually in a browser tab to get the prompt.

**Comment by qlifee** [comment]
> You can try accessing the webdav endpoint manually in a browser tab to get the prompt.

Unfortunately it didn't work, however I managed to get it working on Firefox floccus 5.4.0.

So it might be a problem with chromium based browsers (Chromium, Chrome, Brave).

**Comment by hochl** [comment]
> > You can try accessing the webdav endpoint manually in a browser tab to get the prompt.
> 
> Unfortunately it didn't work, however I managed to get it working on Firefox floccus 5.4.0.
> 
> So it might be a problem with chromium based browsers (Chromium, Chrome, Brave).

I have not been able to sync floccus on any of my Chrome based browsers yet. Maybe my issues are related to me using rclone as WebDav server?

**Comment by marcelklehr** [comment]
> I have not been able to sync floccus on any of my Chrome based browsers yet.

Error message or it didn't happen :)

If it's a network error, you can try connecting to rclone via a different webdav server and check if that works.

**Comment by qlifee** [comment]
> I have the same issue with the same error code E017 on:
> 
> 1. Pixel 9 Android 15 (graphene os) Floccus 5.3.4.
>    Surface running both
> 2. Linux Mint chromium browser Floccus 5.4.0 and
> 3. windows 11 running chrome browser Floccus 5.4.0.
> 
> This is my first time using Floccus and i didn't manage to get it to work.
> 
> I have WEBDAV server up and running on my Synology NAS, and i verified that it is working alright with various applications.
> 
> I suspect the problem is that Floccus doesn't allow self-signed certificates (No option to allow that), because the applications that didn't work are the ones that doesn't allow the self-signed certificates.
> 
> my WebDAV URL is `https://<tailscale-ip-address>:<port>/Bookmarks` File format: XBEL
> 
> Note: I use tailscale to access my NAS.

I am having the same error E017 even when using Nextcloud via Chrome or Chromium browsers, but it works fine with Firefox!

**Comment by marcelklehr** [comment]
So, self-signed certificates are up to the browser. If the browser accepts them, floccus is able to connect, othewise floccus can't do its thing.

**Comment by qlifee** [comment]
I managed to make it work on all Chromium based browsers, and figured out why it worked fine in Firefox.

In Firefox when first you try to access the server `https://<tailscale-ip-address>:<port>/` you'll be prompted to accept the insecure certificate, and once you do that, Firefox will add it in the **server certificate error exception** under **Settings > Privacy & Security > Certificate Manager > Server**

![Screenshot from 2024-12-15 19-40-28](https://github.com/user-attachments/assets/43899e2b-69ac-4caa-89c1-b785905da1e0)


This will ensure the connection will be accepted as an exception.

However in Chromium based browsers this doesn't happen. Instead you'll have to:

1- Create a self-signed certificate like so:

bash
```
openssl req -x509 -nodes -days 365 -newkey rsa:2048 \
-keyout server.key -out server.crt \
-subj "/CN=your-ip-address" \
-addext "subjectAltName=IP:your-ip-address"
```

2- Then combine them into one Base-64 Encoded (PEM Format) certificate:

bash

`cat server.crt server.key > server.pem`

3- Go to Settings > Privacy and Security > Security > Manage certificates > Authorities and Import the .pem certificate.

4- Go to the added certificate starting with org-<tailscale-ip-address> and click the down arrow, then the three dot icon, then click Edit, and check **Trust this certificate for identifying websites**.

![Screenshot from 2024-12-15 19-41-41](https://github.com/user-attachments/assets/bc6c94f2-e701-4cfc-a85c-e0d0b3f525a2)


Now the WebDAV setup with tailscale ip address via floccus will sync, and you won't get the E017 error.

Note: In tailscale settings website make sure to Disable key expiry, this will ensure that this ip address won't be changed automatically.

**Comment by qlifee** [comment]
I also managed to get it to work on Android, I just installed the .pem certificate under CA Certificates and now it works :)

**Comment by marcelklehr** [comment]
> I also managed to get it to work on Android

Oh, then you're the first, I think. Care to share that to the folks at #1560 ?

**Comment by snIP3r23** [comment]
I have the same isse on Win11 with Firefox 133.0.3 with Floccus 5.4.0. But Im using Lets Encrypt Certificates instead of Self Signed. It seems to work with Android an chrome based Kiwi Browser.

**Comment by marcelklehr** [comment]
@snIP3r23 Can you create a new issue in this case, as this one is about self-signed certificates?

**Comment by marcelklehr** [comment]
@hochl Were you able to get it working now?

**Comment by hochl** [comment]
> @hochl Were you able to get it working now?

Hey there! Sorry I am/was busy and will hopefully have time to check it in the new year.

**Comment by grg5811** [comment]
Good morning everyone,
I followed the method form @qlifee but with more detailed instructions found at https://stackoverflow.com/a/60516812 and, now, everything works perfectly.
I work with Browser brave in Debian 12.
Thank you
